cmsutils.py

```diff
@@ -70,6 +70,31 @@ def _memorySizeGB():
     return count
 
 
+def _gpuCount(gpu_type):
+    """
+    Returns the number of supported GPUs for the given type ('cuda' or 'rocm').
+
+    :param gpu_type: Type of GPU to check ('cuda' or 'rocm')
+    :return: Number of supported GPUs
+    """
+    if gpu_type == "cuda":
+        cmd = "cudaComputeCapabilities"
+    elif gpu_type == "rocm":
+        cmd = "rocmComputeCapabilities"
+    else:
+        raise ValueError("Unsupported GPU type: {0}".format(gpu_type))
+
+    e, o = run_cmd(cmd)
+
+    if e != 0:
+        return 0
+
+    supported_lines = [line for line in o.strip().splitlines() if "unsupported" not in line]
+    return len(supported_lines)
```

Comment:
This approach should indeed work most of the time.

More in general, we can have a machine where only some of the GPU are supported; in that case counting the number of supported GPUs may not be enough.
A more general approach would be to return the list of supported GPUs, _e.g._ `[0, 1, 3]` instead of their number.

Reply:
or may be `{cuda|rocm}ComputeCapabilities` already dump this information by adding `supported | unsupported`  key words?

Reply:
> Sure, I can do that.
> 
> But we still need to collect the indices (or ids) of the supported GPUs, we cannot just count how many there are.

sure, if `{cuda|rocm}ComputeCapabilities ` can just dump the gpus ids which we can use then that will help

Reply:
> or may be `{cuda|rocm}ComputeCapabilities` already dump this information by adding `supported | unsupported` key words?

this is already the case, it prints `(unsupported)` for the unsupported ones.

Reply:
> A more general approach would be to return the list of supported GPUs, _e.g._ `[0, 1, 3]` instead of their number.

Thanks, implemented that.